buildSrc/src/test/java/com/osacky/flank/gradle/integration/SanityRoboTest.kt

```diff
@@ -0,0 +1,396 @@
+package com.osacky.flank.gradle.integration
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+
+private fun baseConfigMessage(option: String) = "Incorrect [base] configuration. [$option] can't be used together with sanityRobo."
+private fun additionalConfigMessage(option: String, name: String) = "Incorrect [$name] configuration. [$option] can't be used together with sanityRobo. If you want to launch robo test run without robo script place only sanityRoboRun() into [$name] configuration"
+
+class SanityRoboTest {
+  @get:Rule
+  var testProjectRoot = TemporaryFolder()
+
+  @Before
+  fun setUp() = testProjectRoot.newFile("flank-gradle-service.json").writeText("{}")
+
+  @Test
+  fun `sanityRobo - should throw an error if instrumentationApk set`() {
+    makeBuildDotGradle(
+      where = testProjectRoot,
+      buildScript =
+        """
+      |plugins {
+      |  id "com.osacky.fladle"
+      |}
+      |
+      |fladle {
+      |  debugApk = "debug.apk"
+      |  sanityRobo = true
+      |  serviceAccountCredentials = layout.projectDirectory.file("flank-gradle-service.json")
+      |  instrumentationApk = "test.apk"
+      |  configs {
+      |    sanity {
+      |      sanityRobo.set(true)
+      |      makeSanityRun()
+      |    }
+      |  }
+      |}
+    """
+    )
+
+    val result = failedGradleRun(
+      arguments = listOf("printYml"),
+      projectDir = testProjectRoot.root
+    )
+
+    assertThat(result.output).contains("FAILED")
+    assertThat(result.output).contains(baseConfigMessage("instrumentationApk"))
+
+    val resultSanity = failedGradleRun(
```

Comment:
oh, shouldn't this succeed? or am I missing something?

Reply:
My idea was to take the fail-fast approach and throw an error during the configuration phase

Reply:
Ah makes sense!